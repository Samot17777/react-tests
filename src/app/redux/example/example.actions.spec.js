import * as actions from './example.actions';

describe('Example actions', () => {

    // using snapshoot

    it('creates example action', () => {
        expect(actions.exampleAction(5)).toMatchSnapshot();
    });

    it('creates example action with thunk', () => {
        const dispatch = jest.fn().mockImplementation(x => x);
        const getState = jest.fn().mockImplementation(() => ({
            example: {
                exampleNumber: 5
            }
        }));

        expect(actions.exampleActionUsingThunk(5)(dispatch, getState)).toMatchSnapshot();

        expect(dispatch).toHaveBeenCalled();
        expect(getState).toHaveBeenCalled();
    });

    // using strict comparison

    it('creates example action', () => {
        expect(actions.exampleAction(5)).toEqual({
            payload: 5,
            type: "example/EXAMPLE_ACTION"
        });
    });

    it('creates example action with thunk', () => {
        const dispatch = jest.fn().mockImplementation(x => x);
        const getState = jest.fn().mockImplementation(() => ({
            example: {
                exampleNumber: 5
            }
        }));

        expect(actions.exampleActionUsingThunk(5)(dispatch, getState)).toEqual({
            payload: 10,
            type: "example/EXAMPLE_ACTION"
        });

        expect(dispatch).toHaveBeenCalled();
        expect(getState).toHaveBeenCalled();
    });
});