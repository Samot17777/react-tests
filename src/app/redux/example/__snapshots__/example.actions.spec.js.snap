// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example actions creates example action 1`] = `
Object {
  "payload": 5,
  "type": "example/EXAMPLE_ACTION",
}
`;

exports[`Example actions creates example action with thunk 1`] = `
Object {
  "payload": 10,
  "type": "example/EXAMPLE_ACTION",
}
`;
